sdk/aqueduct/utils/utils.py

```diff
@@ -104,6 +110,16 @@ def parse_user_supplied_id(id: Union[str, uuid.UUID]) -> str:
     return id
 
 
+def parse_user_supplied_id_as_uuid(id: Union[str, uuid.UUID]) -> uuid.UUID:
```

Comment:
nit: Can we name the other helper symmetrically with `parse_..._as_str()`  then?